web/src/helpers/fetch.ts

```diff
@@ -3,29 +3,40 @@
  */
 
 import { Post } from "../types/post.type";
+import { User } from "../types/user.type";
 import { getAuthSession } from "../utils/storage";
 import doRequest from "./request";
 
-interface Params {
-  search?: string;
-  userId?: string | "me";
-  collectionId?: string;
-  limit?: number | string;
-  cursorId?: string;
-}
-
 export async function searchPosts({
   search = "",
   userId = "",
   collectionId = "",
-  limit = "",
+  limit = 16,
   cursorId = "", // cursorId
-}: Params) {
-  const myUserId = getAuthSession().userId;
+}) {
   return await doRequest<{ posts: Post[]; nextCursorId: string }>(
-    `/api/post/search?userId=${myUserId}&search=${search}&searchUserId=${
-      userId === "me" ? myUserId : userId
-    }&collectionId=${collectionId}&limit=${limit}&cursorId=${cursorId}`,
+    `/api/post/search?userId=${
+      getAuthSession().userId
+    }&search=${search}&searchUserId=${userId}&collectionId=${collectionId}&limit=${limit}&cursorId=${cursorId}`,
```

Comment:
Use the object form of [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) to clean this up.